internal/domain_logic/cluster.go

```diff
@@ -93,14 +98,77 @@ func (l *Logic) GetCluster(
 	return cluster, err
 }
 
-func clusterStatus(ufizziCluster *v1alpha1.UffizziCluster) bool {
+func (l *Logic) PatchCluster(
+	clusterName string,
+	namespaceName string,
+	patchClusterParams types.PatchClusterParams,
+) error {
+	namespace, err := l.KuberClient.FindNamespace(namespaceName)
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("namespace/%s found", namespace.Name)
+
+	err = l.KuberClient.PatchCluster(
+		clusterName,
+		namespaceName,
+		patchClusterParams,
+	)
+
+	if err != nil {
+		log.Printf("ClusterError: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+func clusterReadyStatus(ufizziCluster *v1alpha1.UffizziCluster) bool {
 	if len(ufizziCluster.Status.Conditions) == 0 {
 		return false
 	}
 
-	if ufizziCluster.Status.Conditions[0].Status == "False" {
+	readyCondition, err := getConditionByName(ufizziCluster.Status.Conditions, "Ready")
+
+	if err != nil {
+		log.Printf("ClusterError: %s", err)
+		return false
+	}
+
+	if readyCondition.Status == "False" {
 		return false
 	} else {
 		return true
 	}
 }
+
+func clusterSleepStatus(ufizziCluster *v1alpha1.UffizziCluster) bool {
+	if len(ufizziCluster.Status.Conditions) == 0 {
+		return false
+	}
+
+	sleepCondition, err := getConditionByName(ufizziCluster.Status.Conditions, "Sleep")
+
+	if err != nil {
+		log.Printf("ClusterError: %s", err)
+		return false
+	}
+
+	if sleepCondition.Status == "False" {
+		return false
+	} else {
+		return true
+	}
```

Comment:
Can we do just `return !sleepCondition.Status == "False"`
Or maybe `return sleepCondition.Status == "True"`